src/pdn/src/grid.cpp

```diff
@@ -1260,12 +1260,28 @@ ShapeTreeMap InstanceGrid::getInstancePins(odb::dbInst* inst)
     if (net != nullptr) {
       for (auto* mpin : iterm->getMTerm()->getMPins()) {
         for (auto* box : mpin->getGeometry()) {
-          odb::Rect box_rect = box->getBox();
-          transform.apply(box_rect);
-          auto shape
-              = std::make_shared<Shape>(box->getTechLayer(), net, box_rect);
-          shape->setShapeType(Shape::FIXED);
-          pins.push_back(shape);
+          if (box->isVia()) {
+            odb::dbTechVia* tech_via = box->getTechVia();
+            if (tech_via == nullptr) {
+              continue;
+            }
```

Comment:
What is the have a VIA from the LEF itself and not the tech?

Reply:
Would that manifest as a block via? I think there only those two, correct?

Reply:
I see that in lefin any via would become a tech via.  So I guess it isn't possible to have a block via in a macro pin.

Yes there are only block & tech vias.